tests/posix/common/src/pthread.c

```diff
@@ -371,6 +377,67 @@ ZTEST(pthread, test_pthread_termination)
 	zassert_equal(ret, ESRCH, "cancelled a terminated thread!");
 }
 
+ZTEST(pthread, test_pthread_tryjoin)
+{
+	pthread_t th = {0};
+	void *retval;
+
+	/* Creating a threads that exits after 1s*/
+	zassert_ok(pthread_create(&th, NULL, timedjoin_thread, NULL));
+	usleep(USEC_PER_MSEC * 100U);
+
+	/* Attempting to join immediately should fail */
+	zassert_equal(pthread_tryjoin_np(th, &retval), EBUSY);
+
+	/* Sleep so thread will exit */
+	usleep(USEC_PER_MSEC * 200U);
+
+	/* Attempting to join immediately should succeed now */
+	zassert_ok(pthread_tryjoin_np(th, &retval));
+}
+
+ZTEST(pthread, test_pthread_timedjoin)
+{
+	pthread_t th = {0};
+	void *ret;
+	struct timespec not_done;
+	struct timespec done;
+	struct timespec invalid[] = {
+		[0] = {.tv_sec = -1},
+		[1] = {.tv_nsec = -1},
+		[2] = {.tv_nsec = NSEC_PER_SEC},
+	};
+
+	/* setup timespecs when the thread is still running and when it is done */
+	clock_gettime(CLOCK_MONOTONIC, &not_done);
+	clock_gettime(CLOCK_MONOTONIC, &done);
+	not_done.tv_nsec += 100 * NSEC_PER_USEC;
+	done.tv_nsec += 300 * NSEC_PER_USEC;
```

Comment:
@ClaCodes - it looks as though the 100 us and 300 us timeouts do not align with the 100 ms and 300 ms timeouts above.

```suggestion
	not_done.tv_nsec += 100 * NSEC_PER_MSEC;
	done.tv_nsec += 300 * NSEC_PER_MSEC;
```